### PR TITLE
Core/Script Implementation LFG Solo system

### PR DESCRIPTION
**Pull request by zTerragor**
Need test. fix

### CODE REVIEW

File: src/server/worldserver/worldserver.conf.dist [hunk 1]
```diff
@@ -3963,3 +3963,22 @@ Eluna.TraceBack = false
...
 ###############################################################################
+###################################################################################################
+# SOLO LFG
+#
+#    SoloLFG.Enable
+#        Description: Enable the module.
+#        Default:     1  - (Enabled)
+#                     0  - (Disabled)
+
+SoloLFG.Enable = 1
```
Comment (leelf00) [inline src/server/worldserver/worldserver.conf.dist:3974]: Maybe default to Disabled is better.

File: src/server/game/DungeonFinding/LFGQueue.cpp [hunk 2]
```diff
@@ -32,791 +32,444 @@
...
+            if (std::string::npos != itr->second.bestCompatible.find(sguid))
+            {
+                itr->second.bestCompatible.clear();
+                FindBestCompatibleInQueue(itr);
+            }
         }
-    };
-
-    // Readd old queuers with new roles
-    m_tanks.clear();
```
Comment (leelf00) [inline src/server/game/DungeonFinding/LFGQueue.cpp:180]: Does new scripts not check player role?Need test.

Comment (zTerragor) [reply src/server/game/DungeonFinding/LFGQueue.cpp:180]: yes it need test

File: src/server/worldserver/worldserver.conf.dist (hunk 1 above)
Comment (zTerragor) [reply src/server/worldserver/worldserver.conf.dist:3974]: +

File: src/server/scripts/Custom/LFG_Solo.cpp [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+{
+public:
+    lfg_solo_announce() : PlayerScript("lfg_solo_announce") {}
+
+    void OnLogin(Player* player, bool /*firstLogin*/) override
+    {
+        // Announce Module
+        if (sConfigMgr->GetBoolDefault("SoloLFG.Announce", true))
+        {
+            ChatHandler(player->GetSession()).SendSysMessage("This server is running |cff4CFF00Solo Dungeon Finder|r.");
```
Comment (leelf00) [inline src/server/scripts/Custom/LFG_Solo.cpp:27]: Maybe move to trinity_string is better.

File: src/server/game/DungeonFinding/LFGQueue.cpp (hunk 2 above)
Comment (zTerragor) [reply src/server/game/DungeonFinding/LFGQueue.cpp:180]: > 

dont work after test.

File: src/server/scripts/Custom/LFG_Solo.cpp (hunk 3 above)
Comment (fuzionproject) [reply src/server/scripts/Custom/LFG_Solo.cpp:27]: yep

**Pull request closed without merging**